### PR TITLE
Add support to automatically read the container ID within containers

### PR DESCRIPTION
**Pull request by nadundesilva**

### CODE REVIEW

File: choreo-extension-native/src/main/java/io/ballerina/observe/choreo/client/ChoreoClientHolder.java [hunk 1]
```diff
@@ -136,31 +141,56 @@ public static synchronized ChoreoClient getChoreoClient(AutoCloseable dependentO
...
         }
 
-        return instanceId;
+        // Generating new random node ID and setting it into ~/.config/choreo/nodeId file
+        String nodeId = UUID.randomUUID().toString();
+        try {
+            Path nodeIdConfigFileParentPath = nodeIdConfigFilePath.getParent();
+            if (nodeIdConfigFileParentPath == null || nodeIdConfigFileParentPath.toFile().exists()
+                    || nodeIdConfigFileParentPath.toFile().mkdirs()) {
+                Files.write(nodeIdConfigFilePath, nodeId.getBytes(StandardCharsets.UTF_8));
```
Comment (tharinduwijewardane) [inline choreo-extension-native/src/main/java/io/ballerina/observe/choreo/client/ChoreoClientHolder.java:186]: Why do we write if the parent is null (first or condition in above `if`) ?

Comment (tharinduwijewardane) [reply choreo-extension-native/src/main/java/io/ballerina/observe/choreo/client/ChoreoClientHolder.java:186]: Null check should be removed from there, and do it separately as an error handling, if required.

Comment (nadundesilva) [reply choreo-extension-native/src/main/java/io/ballerina/observe/choreo/client/ChoreoClientHolder.java:186]: `nodeIdConfigFilePath.getParent()` can return null if it doesn't have a parent. While this doesn't happen generally the spotbugs checks fails without this check.

```
            if (nodeIdConfigFileParentPath == null || nodeIdConfigFileParentPath.toFile().exists()
                     || nodeIdConfigFileParentPath.toFile().mkdirs()) {
```
This first checks if the config file has a parent. If true checks if the file exists. If it does not it creates the folder and all it's parent directories.
* If this file has no parent, then we can directly create the file.
* If this file's parent exists we do not need to create the parent directory.
It's correct right ? 🤔

Comment (tharinduwijewardane) [reply choreo-extension-native/src/main/java/io/ballerina/observe/choreo/client/ChoreoClientHolder.java:186]: ok

**Review by tharinduwijewardane — APPROVED** [review]

**Pull request merged**